File: back_chat/src/main/java/com/sovereingschool/back_chat/Models/CursoChat.java
package com.sovereingschool.back_chat.Models;

import java.io.Serializable;
import java.util.List;

import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Document(collection = "courses_chat")
public class CursoChat implements Serializable {
    @Id
    private String id;

    private Long idCurso;

    private List<ClaseChat> clases;

    private List<String> mensajes;

}

File: front/src/app/models/CursoChat.ts
import { ClaseChat } from './ClaseChat';
import { MensajeChat } from './MensajeChat';

export class CursoChat {
	id_curso: number;

	clases: ClaseChat[];

	mensajes: MensajeChat[];

	nombre_curso: string;

	foto_curso: string;

	constructor(_id_curso: number, _clases: ClaseChat[], _mensajes: MensajeChat[], _nombre_curso: string, _foto_curso: string) {
		this.id_curso = _id_curso;
		this.clases = _clases;
		this.mensajes = _mensajes;
		this.nombre_curso = _nombre_curso;
		this.foto_curso = _foto_curso;
	}
}
